feat: add support for AUTH_GOOGLE_ALLOWED_DOMAINS environment variabl…

aablsk opened the pull request:
test cloud build

aablsk committed:
feat: add support for AUTH_GOOGLE_ALLOWED_DOMAINS environment variable to restrict Google sign-in to specific domains

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/langfuse/langfuse?pullRequest=1199) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/langfuse/langfuse?pullRequest=1199) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/langfuse/langfuse?pullRequest=1199) it.</sub>

aablsk committed:
feat: add Cloud Build configuration for building and deploying the Langfuse app

This commit adds a Cloud Build configuration file that specifies the steps for building and deploying the Langfuse app. The steps include building the Docker image, pushing it to a container registry, and deploying it to Google Cloud Run. The image that is built and deployed is "europe-west1-docker.pkg.dev/langfuse-test/langfuse/langfuse:latest".

aablsk committed:
chore: add logging option for cloud logging only

aablsk committed:
chore: update Cloud Run deployment with new image using 'services update' instead of 'run deploy' command

aablsk committed:
chore: update build and push steps for Langfuse image

The build and push steps for the Langfuse image have been re-introduced.

marcklingen commented:
Thanks for the contribution! I'd suggest to generalize this to all authentication methods, e.g. `ALLOWLISTED_DOMAINS`. What do you think? Is there anything specific to google auth here?

Can you remove the cloudbuild file?

aablsk commented:
Dear @marcklingen, apologies for this **unintended PR**! 🙇 
I'm testing new git tooling which seems to have opened this PR against the incorrect repository. I'm closing this PR.

I will look into adding a general allowlisting parameter for all providers, but I haven't evaluated what the secure way is for domain verification for other providers, yet, (e.g. it seems that using `email` to verify a domain for Google provider is not recommended) and create a new PR once I've had the time and I'm feeling confident about the solution. 

Unfortunately, I'm currently lacking a way to verify Azure AD as I do not have access to an Azure AD instance with a verified domain.

marcklingen commented:
No worries and thanks for your elaborate answer on this!

I think generalized allow-listing of domains would already be an improvement / useful feature and verification of emails on Google and Azure could be done in separate PRs.
